[Merged by Bors] - Document `Size` and `UiRect`

KDecay opened the pull request:
# Objective

- Migrate changes from #3503.

## Solution

- Document `Size` and `UiRect`.
- I also removed the type alias from the `size_ops` test since it's unnecessary.

## Follow Up

After this change is merged I'd follow up with removing the generics from `Size` and `UiRect` since `Val` should be extensible enough. This was also discussed and decided on in #3503. let me know if this is not needed or wanted anymore!

alice-i-cecile reviewed (approved):
These are excellent. Ping @Weibye @TimJentzsch for a review or two :)

alice-i-cecile commented on crates/bevy_ui/src/geometry.rs:
```diff
@@ -2,17 +2,210 @@ use bevy_math::Vec2;
+///
+/// A margin is used to create space around UI elements, outside of any defined borders.
+///
+/// In this example we are creating a UI margin. It has a left value of 10px, a right value of 20px,
+/// a top value of 30px and a bottom value of 40px. This would add a margin of 10px on the left,
+/// 20px on the right, 30px on the top and 40px on the bottom of the UI element.
```
Agreed; I think we can cut this in favor of letting users read the code.

KDecay replied:
Yup makes sense. The maintainability of this is not the greatest.

KDecay commented:
> For some examples, all the values for left, right, top and bottom are mentioned again at the text, even though they are already specified clearly in the code. I'm wondering if that is necessary, i.e. if it adds additional value. If not I'd prefer to leave it out in favor of shorter docs. I'd ve curious to hear @alice-i-cecile's and @Weibye's opinion on this.

Removed!

> All of the examples are with pixel units, it might make sense to also include some with percentages (or other units if we have more), to demonstrate that this is possible.

I changed some of the `Val` types from `Px` to `Auto` or `Percent` where it made sense. Most of the time `Px` values are used, but I found some nice examples like `UiRect::all(Val::Auto)` for margins which centers the UI element. While searching for some example usage of `padding` in bevy I couldn't find a single example using it so I left it as is for now (seems to be a popular setting lol).

alice-i-cecile reviewed (approved):
bors r+
